itests/docker/docker.go

```diff
@@ -175,14 +179,27 @@ func (d *Docker) Finish(cancel context.CancelFunc) {
 	cancel()
 }
 
+func (d *Docker) stopContainer(containerID string) error {
+	const shutdownTimeout = 5 // In seconds.
+	if stErr := d.pool.Client.StopContainer(containerID, shutdownTimeout); stErr != nil {
+		if klErr := d.pool.Client.KillContainer(dc.KillContainerOptions{
+			ID:     containerID,
+			Signal: dc.SIGINT,
+		}); klErr != nil {
+			return errors.Wrapf(stderrs.Join(stErr, klErr), "failed to stop container %q", containerID)
+		}
+	}
+	return nil
+}
+
 func (d *Docker) startNode(
 	ctx context.Context, cfg config.DockerConfigurator, logFilename, errFilename string,
 ) (*NodeContainer, error) {
 	opts := cfg.DockerRunOptions()
 	opts.Networks = []*dockertest.Network{d.network}
 
 	res, err := d.pool.RunWithOptions(opts, func(hc *dc.HostConfig) {
-		hc.AutoRemove = true
+		hc.AutoRemove = false
```

Comment:
I can't find explicit container removal in `stopContainer` method.
I think it's necessary to set this flag to `true` because it allows docker to clean up container resources.

Reply:
Okay, I see container removal in `NewDocker` constructor. It cleans the previous run of the test suite. Correct me if I'm wrong.

Reply:
And yes, the lost containers removed on start of next tests. I've checked.